// @sdkwork/sdk-common/dist/utils.cjs
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const require_types = require("./core/types.cjs");
const require_logger = require("./utils/logger.cjs");
const require_cache = require("./utils/cache.cjs");
const require_retry = require("./utils/retry.cjs");
const require_string = require("./utils/string.cjs");
const require_encoding = require("./utils/encoding.cjs");
const require_date = require("./utils/date.cjs");
exports.BACKSLASH = require_string.BACKSLASH;
exports.CARRIAGE_RETURN = require_string.CARRIAGE_RETURN;
exports.ConsoleLogger = require_logger.ConsoleLogger;
exports.DASH = require_string.DASH;
exports.DEFAULT_CACHE_CONFIG = require_types.DEFAULT_CACHE_CONFIG;
exports.DEFAULT_LOGGER_CONFIG = require_types.DEFAULT_LOGGER_CONFIG;
exports.DEFAULT_RETRY_CONFIG = require_types.DEFAULT_RETRY_CONFIG;
exports.DEFAULT_TIMEOUT = require_types.DEFAULT_TIMEOUT;
exports.DOT = require_string.DOT;
exports.EMPTY_STRING = require_string.EMPTY_STRING;
Object.defineProperty(exports, "Encoding", {
	enumerable: true,
	get: function() {
		return require_encoding.Encoding;
	}
});
exports.HTTP_STATUS = require_types.HTTP_STATUS;
exports.MILLISECONDS_IN_DAY = require_date.MILLISECONDS_IN_DAY;
exports.MILLISECONDS_IN_HOUR = require_date.MILLISECONDS_IN_HOUR;
exports.MILLISECONDS_IN_MINUTE = require_date.MILLISECONDS_IN_MINUTE;
exports.MILLISECONDS_IN_SECOND = require_date.MILLISECONDS_IN_SECOND;
exports.MILLISECONDS_IN_WEEK = require_date.MILLISECONDS_IN_WEEK;
exports.MIME_TYPES = require_types.MIME_TYPES;
exports.MemoryCacheStore = require_cache.MemoryCacheStore;
exports.NEWLINE = require_string.NEWLINE;
exports.SLASH = require_string.SLASH;
exports.SPACE = require_string.SPACE;
exports.SUCCESS_CODES = require_types.SUCCESS_CODES;
Object.defineProperty(exports, "StringUtils", {
	enumerable: true,
	get: function() {
		return require_string.StringUtils;
	}
});
exports.TAB = require_string.TAB;
exports.TIME_UNITS_IN_MS = require_date.TIME_UNITS_IN_MS;
exports.UNDERSCORE = require_string.UNDERSCORE;
exports.calculateDelay = require_retry.calculateDelay;
exports.createCacheStore = require_cache.createCacheStore;
exports.createLogger = require_logger.createLogger;
exports.createRetryConfig = require_retry.createRetryConfig;
exports.generateCacheKey = require_cache.generateCacheKey;
exports.noopLogger = require_logger.noopLogger;
exports.shouldRetry = require_retry.shouldRetry;
exports.sleep = require_retry.sleep;
exports.withRetry = require_retry.withRetry;
