// @sdkwork/sdk-common/dist/index.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const types = require("./core/types.cjs");
const tokenManager = require("./auth/token-manager.cjs");
const logger = require("./utils/logger.cjs");
const cache = require("./utils/cache.cjs");
const retry = require("./utils/retry.cjs");
require("./utils/string.cjs");
require("./utils/encoding.cjs");
const errors = require("./errors.cjs");
const baseClient = require("./http/base-client.cjs");
exports.DEFAULT_CACHE_CONFIG = types.DEFAULT_CACHE_CONFIG;
exports.DEFAULT_LOGGER_CONFIG = types.DEFAULT_LOGGER_CONFIG;
exports.DEFAULT_RETRY_CONFIG = types.DEFAULT_RETRY_CONFIG;
exports.DEFAULT_TIMEOUT = types.DEFAULT_TIMEOUT;
exports.HTTP_STATUS = types.HTTP_STATUS;
exports.MIME_TYPES = types.MIME_TYPES;
exports.SUCCESS_CODES = types.SUCCESS_CODES;
exports.DefaultAuthTokenManager = tokenManager.DefaultAuthTokenManager;
exports.buildAuthHeaders = tokenManager.buildAuthHeaders;
exports.createTokenManager = tokenManager.createTokenManager;
exports.isTokenValid = tokenManager.isTokenValid;
exports.requiresRefresh = tokenManager.requiresRefresh;
exports.createLogger = logger.createLogger;
exports.noopLogger = logger.noopLogger;
exports.createCacheStore = cache.createCacheStore;
exports.generateCacheKey = cache.generateCacheKey;
exports.calculateDelay = retry.calculateDelay;
exports.createRetryConfig = retry.createRetryConfig;
exports.sleep = retry.sleep;
exports.withRetry = retry.withRetry;
exports.AuthenticationError = errors.AuthenticationError;
exports.CancelledError = errors.CancelledError;
exports.ForbiddenError = errors.ForbiddenError;
exports.NetworkError = errors.NetworkError;
exports.NotFoundError = errors.NotFoundError;
exports.RateLimitError = errors.RateLimitError;
exports.SdkError = errors.SdkError;
exports.ServerError = errors.ServerError;
exports.TimeoutError = errors.TimeoutError;
exports.TokenExpiredError = errors.TokenExpiredError;
exports.ValidationError = errors.ValidationError;
exports.isAuthError = errors.isAuthError;
exports.isNetworkError = errors.isNetworkError;
exports.isRetryableError = errors.isRetryableError;
exports.isSdkError = errors.isSdkError;
exports.BaseHttpClient = baseClient.BaseHttpClient;
exports.createBaseHttpClient = baseClient.createBaseHttpClient;
