// @sdkwork/sdk-common/dist/index.js
import { DEFAULT_CACHE_CONFIG, DEFAULT_LOGGER_CONFIG, DEFAULT_RETRY_CONFIG, DEFAULT_TIMEOUT, HTTP_STATUS, MIME_TYPES, SUCCESS_CODES } from "./core/types.js";
import { DefaultAuthTokenManager, buildAuthHeaders, createTokenManager, isTokenValid, requiresRefresh } from "./auth/token-manager.js";
import { createLogger, noopLogger } from "./utils/logger.js";
import { createCacheStore, generateCacheKey } from "./utils/cache.js";
import { calculateDelay, createRetryConfig, sleep, withRetry } from "./utils/retry.js";
import "./utils/string.js";
import "./utils/encoding.js";
import { AuthenticationError, CancelledError, ForbiddenError, NetworkError, NotFoundError, RateLimitError, SdkError, ServerError, TimeoutError, TokenExpiredError, ValidationError, isAuthError, isNetworkError, isRetryableError, isSdkError } from "./errors.js";
import { BaseHttpClient, createBaseHttpClient } from "./http/base-client.js";
export {
  AuthenticationError,
  BaseHttpClient,
  CancelledError,
  DEFAULT_CACHE_CONFIG,
  DEFAULT_LOGGER_CONFIG,
  DEFAULT_RETRY_CONFIG,
  DEFAULT_TIMEOUT,
  DefaultAuthTokenManager,
  ForbiddenError,
  HTTP_STATUS,
  MIME_TYPES,
  NetworkError,
  NotFoundError,
  RateLimitError,
  SUCCESS_CODES,
  SdkError,
  ServerError,
  TimeoutError,
  TokenExpiredError,
  ValidationError,
  buildAuthHeaders,
  calculateDelay,
  createBaseHttpClient,
  createCacheStore,
  createLogger,
  createRetryConfig,
  createTokenManager,
  generateCacheKey,
  isAuthError,
  isNetworkError,
  isRetryableError,
  isSdkError,
  isTokenValid,
  noopLogger,
  requiresRefresh,
  sleep,
  withRetry
};
