// @sdkwork/sdk-common/dist/utils.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const logger = require("./utils/logger.cjs");
const cache = require("./utils/cache.cjs");
const retry = require("./utils/retry.cjs");
const types = require("./core/types.cjs");
const string = require("./utils/string.cjs");
const encoding = require("./utils/encoding.cjs");
const date = require("./utils/date.cjs");
exports.ConsoleLogger = logger.ConsoleLogger;
exports.createLogger = logger.createLogger;
exports.noopLogger = logger.noopLogger;
exports.MemoryCacheStore = cache.MemoryCacheStore;
exports.createCacheStore = cache.createCacheStore;
exports.generateCacheKey = cache.generateCacheKey;
exports.calculateDelay = retry.calculateDelay;
exports.createRetryConfig = retry.createRetryConfig;
exports.shouldRetry = retry.shouldRetry;
exports.sleep = retry.sleep;
exports.withRetry = retry.withRetry;
exports.DEFAULT_CACHE_CONFIG = types.DEFAULT_CACHE_CONFIG;
exports.DEFAULT_LOGGER_CONFIG = types.DEFAULT_LOGGER_CONFIG;
exports.DEFAULT_RETRY_CONFIG = types.DEFAULT_RETRY_CONFIG;
exports.DEFAULT_TIMEOUT = types.DEFAULT_TIMEOUT;
exports.HTTP_STATUS = types.HTTP_STATUS;
exports.MIME_TYPES = types.MIME_TYPES;
exports.SUCCESS_CODES = types.SUCCESS_CODES;
exports.BACKSLASH = string.BACKSLASH;
exports.CARRIAGE_RETURN = string.CARRIAGE_RETURN;
exports.DASH = string.DASH;
exports.DOT = string.DOT;
exports.EMPTY_STRING = string.EMPTY_STRING;
exports.NEWLINE = string.NEWLINE;
exports.SLASH = string.SLASH;
exports.SPACE = string.SPACE;
Object.defineProperty(exports, "StringUtils", {
  enumerable: true,
  get: () => string.StringUtils
});
exports.TAB = string.TAB;
exports.UNDERSCORE = string.UNDERSCORE;
Object.defineProperty(exports, "Encoding", {
  enumerable: true,
  get: () => encoding.Encoding
});
exports.MILLISECONDS_IN_DAY = date.MILLISECONDS_IN_DAY;
exports.MILLISECONDS_IN_HOUR = date.MILLISECONDS_IN_HOUR;
exports.MILLISECONDS_IN_MINUTE = date.MILLISECONDS_IN_MINUTE;
exports.MILLISECONDS_IN_SECOND = date.MILLISECONDS_IN_SECOND;
exports.MILLISECONDS_IN_WEEK = date.MILLISECONDS_IN_WEEK;
exports.TIME_UNITS_IN_MS = date.TIME_UNITS_IN_MS;
